### PR TITLE
feat: add --ratchet flag for regression-only CI failures

### PR DESCRIPTION
**Pull request by sebastianbreguel**
## Summary

- Adds `--ratchet` / `-R` boolean flag that combines with `--diff <ref>`
- Changes exit code behavior: CI only fails when complexity **regressed** (REGRESSED functions or NEW functions exceeding `--max-complexity-allowed`)
- Pre-existing complex code passes — enables incremental adoption without fixing legacy debt first
- Supports TOML config: `ratchet = true`
- Validates that `--ratchet` requires `--diff` (exit 2 otherwise)

### Usage

```bash
# Only fail if complexity got worse vs main
complexipy . --diff main --ratchet

# In pyproject.toml
[tool.complexipy]
ratchet = true
```

## Test plan

- [x] 10 unit tests for `has_regressions()` covering all statuses and edge cases
- [x] Full test suite passes (56 passed)
- [x] Pre-commit hooks pass
- [x] Manual CLI verification: exit 1 on regression, exit 0 on pre-existing complex code, exit 2 on `--ratchet` without `--diff`

### CODE REVIEW

**Comment by rohaquinlop** [comment]
Hi @sebastianbreguel which is the main difference between this new parameter and the snapshots?

**Comment by sebastianbreguel** [comment]
Great question! The key differences:

**Snapshots** require creating and committing a `complexipy-snapshot.json` file that records the baseline. You maintain this file over time, and complexipy compares against it locally.

**Ratchet** (`--ratchet --diff <ref>`) needs no extra files — it compares directly against a git ref (e.g., `main`). The baseline is always the current state of the target branch, so it stays up to date automatically.

In practice:
- **Snapshots** = explicit, file-based baseline you manage manually
- **Ratchet** = implicit, git-based baseline that updates as `main` evolves

Both solve the same core problem (incremental adoption without fixing legacy debt), but ratchet is zero-maintenance since it piggybacks on `--diff` and git history rather than a separate artifact.

They can also complement each other: snapshots for tracking absolute thresholds over time, ratchet for CI gating that only blocks regressions per-PR.

File: complexipy/utils/diff.py [hunk 1]
```diff
@@ -178,6 +178,25 @@ def _build_diff_summary(changed: List[DiffEntry]) -> str:
...
 
+def has_regressions(entries: List[DiffEntry], max_complexity: int) -> bool:
+    """Return True if any entry represents a ratchet failure.
+
+    A ratchet failure is:
+    - A REGRESSED function (complexity increased vs the reference)
+    - A NEW function whose complexity exceeds *max_complexity*
+    """
+    for e in entries:
+        if e.status == _STATUS_REGRESSED:
```
Comment (rohaquinlop) [inline complexipy/utils/diff.py]: This makes `--ratchet` fail on any complexity increase, even when the function is still under `--max-complexity-allowed` (for example `3 -> 4` with threshold `15`). That feels stricter than the rest of `complexipy`, where the threshold is the main contract. I think ratchet should fail modified existing functions only when they both regress and end up above the threshold, while keeping the current behavior for `NEW` functions above threshold. Otherwise CI will block low-complexity changes that are not actually violations.

**Review by rohaquinlop — CHANGES_REQUESTED** [review]
Please include `--ratchet` to the project docs, and explain when and why it's useful to use, this way we help new users to use this parameter

**Comment by sebastianbreguel** [comment]
Good catch — fixed in 33ed20b.

`has_regressions` now flags a REGRESSED entry only when `new_complexity > max_complexity`, so modified functions that increase but stay within `--max-complexity-allowed` no longer break CI (e.g. `3 → 4` with threshold `15` passes). New functions above the threshold still fail, and already-over functions that get worse also fail.

Also added a **Ratchet Mode** section to `docs/usage-guide.md` and `docs/index.md` (plus the Spanish versions) covering when/why to use it — incremental cleanup of legacy codebases, CI gates that only fire on real threshold breaches, and gradual `--max-complexity-allowed` tightening. The `--ratchet` flag is now listed in the CLI reference table, and the `typer` help text reflects the new semantics.

**Review by rohaquinlop — APPROVED** [review]
LGTM! thank you so much for your contributions ❤️
